src/deps_log.cc

```diff
@@ -225,8 +225,9 @@ LoadStatus DepsLog::Load(const string& path, State* state, string* err) {
       int* deps_data = reinterpret_cast<int*>(buf);
       int out_id = deps_data[0];
       TimeStamp mtime;
-      mtime = (TimeStamp)(((uint64_t)(unsigned int)deps_data[2] << 32) |
-                          (uint64_t)(unsigned int)deps_data[1]);
+      mtime = static_cast<TimeStamp>(
+          static_cast<uint64_t>(static_cast<unsigned int>(deps_data[2])) << 32 |
+          static_cast<uint64_t>(static_cast<unsigned int>(deps_data[1])));
```

Comment:
This whole series of logic is awful.

Can it be cleaned up, or at the very least, documented to explain what the intention is?

Reply:
No idea what this code does.

Reply:
Sure. That's fair. Semantically you're not changing anything, so I guess it's no worse than before.

Reply:
Could be.

Makes me wonder if it should be a call to ntoh or hton instead?